client_code/storage.py

```diff
@@ -99,43 +117,129 @@ def pop(self, key, default=None):
 
     def clear(self):
         """clear all items from local storage"""
-        for key in self._filter_store():
-            self._store.removeItem(key)
+        self._store.clear()
 
     def update(self, other, **kws):
         """update the local storage item with key/value pairs from other"""
         other = dict(other, **kws)
         for key, value in other.items():
             self[key] = value
 
-
-local_storage = Storage(_window.get("localStorage"))
-session_storage = Storage(_window.get("sessionStorage"))
+    def __eq__(self, other):
+        if not isinstance(other, StorageWrapper):
+            return NotImplemented
+        return self._driver == other._driver and self._name == other._name
+
+    def get_store(self, name):
```

Comment:
This kind of factory function might be better as either a class method or a module level function - named something like 'create_store'

Reply:
Yeah - i think i'd prefer it to be a `classmethod` too. 
I just struggled to think of the best way to do that since I needed to know if you wanted an `indexed_db` store or a `local_storage` store.

How about - I subclass the `StorageWrapper` into `LocalStorageWrapper` and `IndexedDBStorageWrapper`

Yes That'd work. And it also makes things a bit clearer. `indexed_db` is just the default store instance of the `IndexedDBStorageWrapper`.

And @meatballs votes for `create_store`. Since that was your instinct for this method lets go with that.

Reply:
The other alternative is to use a module level factory function with the 'storage_type' as an argument. You then don't need the subclasses.

Reply:
(for me, inheritance is one of those situations where you sit on your hands for a moment and ponder whether there might be another way)!

Reply:
I can see that -  what would you vote the type of the `storage_type` to be?

Reply:
I'd make that a simple string parameter and either map the string to the relevant class within the function or use `getattr` and the relevant module to derive the class.

Reply:
I'd make that a string and map it to either `_indexed_db_wrapper` or `_local_storage_wrapper` within the function's body.

Reply:
or, I suppose, we could choose one of them to be the default (say, indexed_db) and use a boolean like `use_local_storage` with a default of `False`

Reply:
I'll sleep on it - I already wrote the inheritance version before I read:
> inheritance is one of those situations where you sit on your hands for a moment and ponder whether there might be another way

So i've pushed that for now: 0aa8577151d204542378cccddd54173b5a85ba12
Maybe take a look at it if you get a moment. See how much it makes you wince...

Reply:
It's absolutely fine. I'd have probably gone with a module level function, but the difference is marginal.

You done with this now? It looks ready to merge to me.

Reply:
I did a little clean up yesterday and played around with the module level function. (not yet commited)

Let's call it approved - and then i'll hit the merge button later today. (when i need a break from javascript)